src/ed25519.rs

```diff
@@ -110,7 +111,7 @@ impl PartialEq for Signature {
         let mut equal: u8 = 0;
 
         for i in 0..32 {
-            equal |= self.r.0[i] ^ other.r.0[i];
+            equal |= self.R.0[i] ^ other.R.0[i];
             equal |= self.s[i]   ^ other.s[i];
```

Comment:
hmm, this isn't related to this change, but why doesn't `Signature` just `#[derive(Eq, PartialEq)]`?